bindgen/codegen/mod.rs

```diff
@@ -3450,128 +3416,211 @@ impl<'a> EnumBuilder<'a> {
+                    value: quote! { #expr },
                 });
 
                 self
             }
-            EnumBuilder::ModuleConsts {
-                module_name,
-                mut module_items,
-            } => {
+            EnumBuilderVariant::ModuleConsts { .. } => {
                 let name = ctx.rust_ident(variant_name);
-                let ty = ctx.rust_ident(CONSTIFIED_ENUM_MODULE_REPR_NAME);
-                module_items.push(quote! {
-                    #doc
-                    pub const #name : #ty = #expr ;
+                enum_variants.push(EnumVariantInfo {
+                    variant_name: name,
+                    variant_doc,
+                    value: quote! { #expr },
                 });
+                self
+            }
+        }
+    }
 
-                EnumBuilder::ModuleConsts {
-                    module_name,
-                    module_items,
-                }
+    fn newtype_bitfield_impl(
+        prefix: Ident,
+        rust_ty: &syn::Type,
+    ) -> proc_macro2::TokenStream {
+        let rust_ty_name = &rust_ty;
+        quote! {
+        impl ::#prefix::ops::BitOr<#rust_ty> for #rust_ty {
+            type Output = Self;
+
+            #[inline]
+            fn bitor(self, other: Self) -> Self {
+                #rust_ty_name(self.0 | other.0)
+            }
+        }
+        impl ::#prefix::ops::BitOrAssign for #rust_ty {
+            #[inline]
+            fn bitor_assign(&mut self, rhs: #rust_ty) {
+                self.0 |= rhs.0;
+            }
+        }
+        impl ::#prefix::ops::BitAnd<#rust_ty> for #rust_ty {
+            type Output = Self;
+
+            #[inline]
+            fn bitand(self, other: Self) -> Self {
+                #rust_ty_name(self.0 & other.0)
+            }
+        }
+        impl ::#prefix::ops::BitAndAssign for #rust_ty {
+            #[inline]
+            fn bitand_assign(&mut self, rhs: #rust_ty) {
+                self.0 &= rhs.0;
             }
         }
+        }
     }
 
     fn build(
         self,
         ctx: &BindgenContext,
         rust_ty: &syn::Type,
-        result: &mut CodegenResult<'_>,
+        enum_variants: &[EnumVariantInfo],
     ) -> proc_macro2::TokenStream {
-        match self {
-            EnumBuilder::Rust {
-                attrs,
-                ident,
-                tokens,
-                emitted_any_variants,
-                ..
-            } => {
-                let variants = if emitted_any_variants {
-                    tokens
-                } else {
-                    quote!(__bindgen_cannot_repr_c_on_empty_enum = 0)
-                };
+        let mut variant_idents = vec![];
+        let mut variant_values = vec![];
+        let mut variant_docs = vec![];
+        for v in enum_variants {
+            variant_idents.push(v.variant_name.clone());
+            variant_values.push(v.value.clone());
+            variant_docs.push(v.variant_doc.clone());
+        }
```

Comment:
I'm confused, are these three vectors just to simplify the generation? If so I'd prefer to just iterate over `enum_variants` and bind to identifiers like:

```
let ident = &v.variant_name;
let value = &v.value;
let doc = &v.variant_doc;
```

Reply:
Yeah, that doesn't make much sense - was a remnant of an older iteration. Changed.